### PR TITLE
Remove N2O ambiguity

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/openscm/openscm-units/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/openscm/openscm-units/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/openscm/openscm-units/issues/YY>`_)
```

### CODE REVIEW

**Review by lewisjared — APPROVED** [review]
The conversion looks good.

Why would one want to do N2O -> N rather than N2O -> N2ON? Is N2ON the prevailing term used for this type of conversion? If the more common conversion is to count 2xN atoms why not make the 1xN conversion the special case.

**Comment by znicholls** [comment]
> Is N2ON the prevailing term used for this type of conversion?

That is my understanding. Of course, it's all ambiguous (hence this PR to try and remove the ambiguity).

> If the more common conversion is to count 2xN atoms why not make the 1xN conversion the special case.

So, if you do N2O --> N, the conversion factor would be 28 / 44 rather than 14 / 44? Could do either (or could just make this always forbidden). I'm a bit unsure what the best approach is.

File: tests/unit/test_units.py [hunk 1]
```diff
@@ -149,32 +154,32 @@ def test_a():
 
 def test_context():
     CO2 = unit_registry("CO2")
-    N = unit_registry("N")
+    N2ON = unit_registry("N2ON")
     with unit_registry.context("AR4GWP100"):
-        np.testing.assert_allclose(CO2.to("N").magnitude, 14 / (44 * 298))
-        np.testing.assert_allclose(N.to("CO2").magnitude, 44 * 298 / 14)
+        np.testing.assert_allclose(CO2.to("N2ON").magnitude, 28 / (44 * 298))
+        np.testing.assert_allclose(N2ON.to("CO2").magnitude, 44 * 298 / 28)
```
Comment (lewisjared) [inline tests/unit/test_units.py:160]: Can you add a test for N2O <-> CO2 to show that N2O and N2ON result in different values

**Review by lewisjared — APPROVED** [review]
After a bit of a google, N2ON does seem to be used somewhat widely. This approach seems reasonable.

File: tests/unit/test_units.py (hunk 1 above)
Comment (znicholls) [reply tests/unit/test_units.py:160]: Done

File: tests/unit/test_units.py [hunk 2]
```diff
@@ -42,8 +49,6 @@ def test_nox():
         np.testing.assert_allclose(N.to("NOx").magnitude, 46 / 14)
         np.testing.assert_allclose(NO2.to("NOx").magnitude, 1)
         np.testing.assert_allclose(NOx.to("NO2").magnitude, 1)
-        # this also becomes allowed, unfortunately...
-        np.testing.assert_allclose(NOx.to("N2O").magnitude, 44 / 46)
```
Comment (mikapfl) [inline tests/unit/test_units.py:46]: Should this be added as a test that it explicitly doesn't work anymore?

**Review by mikapfl — APPROVED** [review]
I have no idea about the nitrogen conversions themselves (i.e. what is typical in emissions reports), but the code looks good.

File: tests/unit/test_units.py (hunk 2 above)
Comment (znicholls) [reply tests/unit/test_units.py:46]: very nice, added

**Pull request merged**